PluginCode/src/main/java/smr/cs/ualberta/libcomp/action/ReplacementAction.java

```diff
@@ -336,58 +364,42 @@ public void detectMaven(@NotNull final Editor editor, @NotNull final PsiFile psi
 
                     MavenListObjects.add(depObj);
 
-
                     if (depObj.getEnableddomain()) {
-                        editorModel.addLineHighlighter(i,
+                        editorModel.addLineHighlighter(i-1,
                                 DebuggerColors.BREAKPOINT_HIGHLIGHTER_LAYER + 1, softerAttributes);
-
                     }
-
-
                 }
-            }
-            ++i;
-            boolean isContainsEnd = lineText.contains("</dependencies>");
-            if (isContainsEnd) {
-                dependenciesExists = false;
+                startOffset = document.getLineStartOffset(i);
+                endOffset = document.getLineEndOffset(i);
+                lineText = document.getText(new TextRange(startOffset, endOffset)).trim();
             }
         }
+        catch (IOException | XmlPullParserException ioException) {
+            ioException.printStackTrace();
+        }
     }
 
-    public void detectMavenOnAction(@NotNull final AnActionEvent event) {
+    public void detectOnAction(@NotNull final AnActionEvent event, @NotNull final String fileType) throws IOException {
```

Comment:
can you add javadoc documentation to the functions in general to document their parameters and purpose?

Reply:
sure